configure.ac

```diff
@@ -363,26 +363,14 @@ if test "x$PRESET_CFLAGS" = "x"; then
   fi
 fi
 
-dnl set squid required flags
-dnl TODO use SQUID_CC_ADD_CXXFLAG_IF_SUPPORTED instead of hardcoding
-if test "$squid_cv_compiler" = "gcc"; then
-  case "$squid_host_os" in
-  mingw)
-dnl Guido Serassio (serassio@squid-cache.org) 20070811
-dnl Using the latest MinGW (gcc 3.4.5 + mingw-runtime 3.13) cannot build with
-dnl -Werror -Wmissing-prototypes -Wmissing-declarations
-dnl TODO: check if the problem will be present in any other newer MinGW release.
-    SQUID_CFLAGS="$SQUID_CFLAGS $squid_cv_cc_option_wall -Wpointer-arith -Wwrite-strings -Wcomments -Wshadow"
-    ;;
-  *)
-    SQUID_CFLAGS="$SQUID_CFLAGS $squid_cv_cc_option_wall -Wpointer-arith -Wwrite-strings -Wmissing-prototypes -Wmissing-declarations -Wcomments -Wshadow"
-    ;;
-  esac
-  SQUID_CXXFLAGS="$SQUID_CXXFLAGS $squid_cv_cc_option_wall -Wpointer-arith -Wwrite-strings -Wcomments -Wshadow -Woverloaded-virtual"
-else
-  SQUID_CFLAGS=${SQUID_CFLAGS:-}
-  SQUID_CXXFLAGS=${SQUID_CXXFLAGS:-}
-fi
+# be very pedantic with warnings, if the compiler supports it
```

Comment:
Let's try to add a potentially useful comment rather than retell what the code clearly does. Please pick the right version or adjust as needed:

```suggestion
# useful warnings that may not be included in -Wall -Wextra
```

or

```suggestion
# useful warnings that are not included in -Wall -Wextra
```